Improves maintenance domain name form element

dstrukt opened the pull request:
More small fixes.

Updated domain form section to flex-box (as opposed to the "row/g-2" & "col-sm" pattern cc @dennisreimann curious to hear your thoughts on this, as we might have some consistencies throughout the app regardless).

The display error below was caused by a stray "mb-2" class, and we could simply adjust and remove that class, if flex-box isn't the path forward.

<img width="517" alt="Screen Shot 2021-10-31 at 5 09 05 PM" src="https://user-images.githubusercontent.com/6250771/139606496-543884eb-bdf4-484b-8e9a-f605f46fa77e.png">

Otherwise, just some spacing and consistency edits.

After:
<img width="1319" alt="Screen Shot 2021-10-31 at 5 50 01 PM" src="https://user-images.githubusercontent.com/6250771/139607721-4d63d267-04ac-4bdc-8e41-1fb2a2055a02.png">
<img width="394" alt="Screen Shot 2021-10-31 at 5 51 40 PM" src="https://user-images.githubusercontent.com/6250771/139607741-0374df26-b3a8-4a00-afe7-c6ec3c43e33d.png">

dennisreimann commented on BTCPayServer/Views/Server/Maintenance.cshtml:
```diff
@@ -8,38 +8,34 @@
-            <h4 class="mb-3">Domain name</h4>
-            <p>You can change the domain name of your server by following <a href="https://docs.btcpayserver.org/Deployment/ChangeDomain/" target="_blank" rel="noreferrer noopener">this guide</a>.</p>
+            <h4 class="mb-2">Domain name</h4>
+            <p class="text-secondary">You can change the domain name of your server by following <a href="https://docs.btcpayserver.org/Deployment/ChangeDomain/" target="_blank" rel="noreferrer noopener">this guide</a>.</pclass="text-secondary">
```
I'd rather remove the `text-secondary` class for the other paragraphs on that page, because we don't use them like that elsewhere. Ok?

dstrukt replied:
Sorry, maybe it's just late and I'm misunderstanding, are you saying we should remove the text-secondary class specifically for this snippet of text?

If so, that's fine with me, I'll make the change in the morning!

dennisreimann replied:
Yes, I'd remove it here and from the other paragraphs on that page as well.

dstrukt commented:
@dennisreimann Latest commit addresses your feedback!

Screenshot:
<img width="577" alt="Screen Shot 2021-11-01 at 12 40 34 PM" src="https://user-images.githubusercontent.com/6250771/139731595-9b536886-aa14-479c-bb13-099679a11456.png">

dennisreimann reviewed (approved):
Nice, that's what I meant! Thank you 👍
